ntbs-service/DataMigration/ImportValidator.cs

```diff
@@ -101,21 +101,43 @@ public ImportValidator(IImportLogger logger, IReferenceDataRepository referenceD
+            TrimAndCleanStringProperties(notification.ClinicalDetails);
+            TrimAndCleanStringProperties(notification.HospitalDetails);
+            TrimAndCleanStringProperties(notification.PreviousTbHistory); // ?
+            TrimAndCleanStringProperties(notification.ComorbidityDetails); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorHomelessness); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorDrugs); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorImprisonment); // ?
+            TrimAndCleanStringProperties(notification.SocialRiskFactors?.RiskFactorSmoking); // ?
+            TrimAndCleanStringProperties(notification.ImmunosuppressionDetails);
+            TrimAndCleanStringProperties(notification.TravelDetails); // ?
+            TrimAndCleanStringProperties(notification.VisitorDetails); // ?
+            TrimAndCleanStringProperties(notification.DenotificationDetails); // untested
+            TrimAndCleanStringProperties(notification.ComorbidityDetails); // ?
+            TrimAndCleanStringProperties(notification.MDRDetails);
+            TrimAndCleanStringProperties(notification.Group); // ? untested
+            TrimAndCleanStringProperties(notification.TestData); // ?
+            TrimAndCleanStringProperties(notification.MBovisDetails); // ?
+            TrimAndCleanStringProperties(notification.DrugResistanceProfile);
+
+            notification.NotificationSites?.ForEach(TrimAndCleanStringProperties); // untested
+            notification.TestData.ManualTestResults.ForEach(TrimAndCleanStringProperties); // ? untested
+            notification.SocialContextAddresses.ForEach(TrimAndCleanStringProperties);
+            notification.SocialContextVenues.ForEach(TrimAndCleanStringProperties);
+            notification.TreatmentEvents.ForEach(TrimAndCleanStringProperties);
+            notification.MBovisDetails.MBovisExposureToKnownCases.ForEach(TrimAndCleanStringProperties);
+            notification.MBovisDetails.MBovisOccupationExposures.ForEach(TrimAndCleanStringProperties);
+            notification.MBovisDetails.MBovisAnimalExposures.ForEach(TrimAndCleanStringProperties);
+            notification.MBovisDetails.MBovisUnpasteurisedMilkConsumptions.ForEach(TrimAndCleanStringProperties);
         }
 
-        private static void CleanContainsNoTabsProperties(object entity)
+        private static void TrimAndCleanStringProperties(object entity)
         {
             entity?.GetType()
                 .GetProperties()
-                .Where(prop => Attribute.IsDefined(prop, typeof(ContainsNoTabsAttribute)))
+                .Where(prop => prop.PropertyType == typeof(string) && prop.CanWrite)
                 .ForEach(prop => prop.SetValue(entity, TrimAndCleanString((string)prop.GetValue(entity))));
```

Comment:
The strategy here now being to find any writable `string` (so calculated `readonly` properties like `LegacyId` don't cause us problems) and clean it, instead of any property with the `[ContainsNoTabs]` attribute

Reply:
Yeah I like it! It misses a few of the fields I was covering before, but these were key-like fields like `TbServiceCode` which already matched the validation, but that's fine and hopefully less surprising to the next person who modifies these classes.